src/client/security.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  HttpScheme,
+  isApiKeySecurityValues,
+  isBasicAuthSecurityValues,
+  isBearerTokenSecurityValues,
+  isDigestSecurityValues,
+  SecurityScheme,
+  SecurityType,
+  SecurityValues,
+} from '@superfaceai/ast';
+
+import {
+  invalidSecurityValuesError,
+  securityNotFoundError,
+} from '../internal/errors.helpers';
+import { SecurityConfiguration } from '../internal/interpreter/http';
+
+export function resolveSecurityConfiguration(
+  schemes: SecurityScheme[],
+  values: SecurityValues[],
+  providerName: string
+): SecurityConfiguration[] {
+  const result: SecurityConfiguration[] = [];
+
+  for (const vals of values) {
+    const scheme = schemes.find(scheme => scheme.id === vals.id);
+    if (scheme === undefined) {
+      const definedSchemes = schemes.map(s => s.id);
+      throw securityNotFoundError(providerName, definedSchemes, vals);
+    }
+
+    const invalidSchemeValuesErrorBuilder = (
+      scheme: SecurityScheme,
+      values: SecurityValues,
+      requiredKeys: [string, ...string[]]
+    ) => {
+      const valueKeys = Object.keys(values).filter(k => k !== 'id');
+
+      return invalidSecurityValuesError(
```

Comment:
should this be `throw`?

Reply:
It's a builder method, defined inline the function.

Reply:
oh yeah, I'm blind